extension/src/recorder.ts

```diff
@@ -57,3 +68,102 @@ window.addEventListener(
   },
   { capture: true, passive: true }
 )
+
+function handleSelectChange(target: HTMLSelectElement) {
+  captureEvents({
+    type: 'select',
+    eventId: crypto.randomUUID(),
+    timestamp: Date.now(),
+    selector: generateSelector(target),
+    selected: [...target.selectedOptions].map((option) => option.value),
+    multiple: target.multiple,
+    tab: '',
+  })
+}
+
+function handleTextAreaChange(target: HTMLTextAreaElement) {
+  captureEvents({
+    type: 'input-change',
+    eventId: crypto.randomUUID(),
+    timestamp: Date.now(),
+    selector: generateSelector(target),
+    value: target.value,
+    tab: '',
+  })
+}
+
+function handleInputChange(target: HTMLInputElement) {
+  if (
+    target.type === 'button' ||
+    target.type === 'submit' ||
+    target.type === 'reset' ||
+    target.type === 'file' ||
+    target.type === 'image'
+  ) {
+    return
+  }
+
+  if (target.type === 'checkbox') {
+    captureEvents({
+      type: 'check',
+      eventId: crypto.randomUUID(),
+      timestamp: Date.now(),
+      selector: generateSelector(target),
+      checked: target.checked,
+      tab: '',
+    })
+
+    return
+  }
+
+  if (target.type === 'radio') {
+    if (!target.checked) {
```

Comment:
For radio buttons there will be two change events: one for the new element being checked and one for the old element that was unchecked. We're only interested in one of them, so we don't generate an `check`/`uncheck` call.

It also gets it's own event because of how radios are grouped on the `name` of the control and that's nice to know from a UI perspective (though maybe that's true of all inputs? 🤔 )